src/sentry/services/hybrid_cloud/organization_actions/impl.py

```diff
@@ -47,14 +47,37 @@ def upsert_organization_by_org_id_with_outbox_message(
 def mark_organization_as_pending_deletion_with_outbox_message(
     *, org_id: int
 ) -> Optional[Organization]:
-    with transaction.atomic():
-        query_result = Organization.objects.filter(
+    with outbox_context(transaction.atomic()):
+        update_count = Organization.objects.filter(
             id=org_id, status=OrganizationStatus.ACTIVE
         ).update(status=OrganizationStatus.PENDING_DELETION)
 
-        if not query_result:
+        if not update_count:
             return None
 
+        assert update_count == 1
```

Comment:
Should we be raising an error if no rows were updated because the record was already in pending delete state? If no update was made we could still emit the outbox to ensure the other silo converges.